app/controllers/tour_pages_controller.rb

```diff
@@ -1,3 +1,7 @@
 class TourPagesController < ApplicationController
-  def home; end
+  def home
+    @tour = TourDetail.includes(:tour)
```

Comment:
`TourDetail.includes(:tour)` sao ngược ngược vậy em, controller Tour mà lại mấy từ model Tourdetail ah? 

Nếu code của em, có cần includes ko thế

giải thích cơ chế N+1 query cho chị nghe nào

Reply:
màn hình list thì thêm xử lý sort vào cho chị nhé

Reply:
đây là list thì tên object nên đạt số nhiều nhé `tours`

Reply:
Dạ controller này em đặt tên chung cho tất cả các tour. Em dùng includes, lấy nhiều TourDetail nối với 1 Tour để lấy hết toàn bộ các tour luôn. 
Cơ chế N+1 query là lúc mình truy vấn thì hệ thống nó phát sinh truy vấn nhiều lần theo từng id. Mình giải quyết bằng cách dùng includes thì truy vấn chỉ cần gộp id và truy vấn 1 lần.
Dạ để em sort theo `start_date` sửa tên biến lại